### PR TITLE
FieldConfig API: Add ability to hide field option or disable it from the overrides.

### PR DESCRIPTION
**Pull request by dprokop**
This PR adds possibility to either:
- hide option from field config UI
- disable possibility to create an option override
- or both of the above

It introduces two new properties to the Field Options API (FieldConfigEditorConfig):
```ts
/** Hides option from the Field config tab */
hideFromDefaults?: boolean;

/** Indicates that option should not be available for the overrides */
hideFromOverrides?: boolean;
```

TODO:
- [ ] Add tests for OverrideEditor.tsx and DefaultFieldConfigEditor.tsx

### CODE REVIEW

File: packages/grafana-data/src/types/fieldOverrides.ts [hunk 1]
```diff
@@ -48,6 +48,12 @@ export interface FieldConfigEditorConfig<TOptions, TSettings = any, TValue = any
    * @param field
    */
   shouldApply?: (field: Field) => boolean;
+
+  /** Indicates that option shoukd not be available in the Field config tab */
+  hideFromFieldOptions?: boolean;
```
Comment (dprokop) [inline packages/grafana-data/src/types/fieldOverrides.ts:53]: Not a fan of this name, happy for any suggestion :)

File: packages/grafana-data/src/types/fieldOverrides.ts [hunk 2]
```diff
@@ -50,10 +50,10 @@ export interface FieldConfigEditorConfig<TOptions, TSettings = any, TValue = any
   shouldApply?: (field: Field) => boolean;
 
   /** Indicates that option shoukd not be available in the Field config tab */
-  hideFromFieldOptions?: boolean;
+  hideFromDefaults?: boolean;
```
Comment (jackw) [inline packages/grafana-data/src/types/fieldOverrides.ts:53]: nit: I'm probably missing some context or a naming convention here but I find the previous name easier to follow.

**Review by jackw — APPROVED** [review]
LGTM! 👍

File: packages/grafana-data/src/types/fieldOverrides.ts (hunk 1 above)
Comment (stevenvachon) [reply packages/grafana-data/src/types/fieldOverrides.ts]: `customizable?: boolean` ?

**Review by torkelo — APPROVED** [review]
Looks good, like the new naming (hideFromDefaults & hideFromOverrides)

File: packages/grafana-data/src/types/fieldOverrides.ts (hunk 2 above)
Comment (dprokop) [reply packages/grafana-data/src/types/fieldOverrides.ts:53]: @jackw to shed some light on this. When we talk about "default field options" we refer to the options that are available from the Fields tab in the panel editor's side pane. This is pretty much a naming convention that we use, hence this suggestion from Torkel.

**Pull request merged**